pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,1502 @@
+        /// Handler for the unbalanced reduction when slashing a staker.
+        type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
+        /// Number of sessions that staked fund remain bonded for
+        type BondedDuration: Get<SessionIndex>;
+        /// Number of sessions that slashes are deferred by, after computation.
+        type SlashDeferDuration: Get<SessionIndex>;
+        /// Minimum number of selected validators every round
+        type MinSelectedValidators: Get<u32>;
+        /// Maximum nominators per validator
+        type MaxNominatorsPerValidator: Get<u32>;
+        /// Maximum validators per nominator
+        type MaxValidatorPerNominator: Get<u32>;
+        /// Fee due to validators, set at genesis
+        type DefaultValidatorFee: Get<Perbill>;
+        /// Default Slash reward propostion, set at genesis
+        type DefaultSlashRewardProportion: Get<Perbill>;
+        /// The proportion of the slashing reward to be paid out on the first slashing detection.
+        type DefaultSlashRewardFraction: Get<Perbill>;
+        /// Minimum stake required for any account to be in `SelectedCandidates` for the round
+        type MinValidatorStake: Get<BalanceOf<Self>>;
+        /// Minimum stake required for any account to be a validator candidate
+        type MinValidatorPoolStake: Get<BalanceOf<Self>>;
+        /// Minimum stake for any registered on-chain account to nominate
+        type MinNomination: Get<BalanceOf<Self>>;
+        /// Minimum stake for any registered on-chain account to become a nominator
+        type MinNominatorStake: Get<BalanceOf<Self>>;
+        /// Tokens have been minted and are unused for validator-reward.
+        /// See [Era payout](./index.html#era-payout).
+        type RewardRemainder: OnUnbalanced<NegativeImbalanceOf<Self>>;
+        /// Interface for interacting with a session module.
+        type SessionInterface: SessionInterface<Self::AccountId>;
+        /// This pallet's module id. Used to derivate a dedicated account id to store session
+        /// rewards for validators and nominators in.
+        type PalletId: Get<ModuleId>;
+        /// staking pallet Lock Identifier used for set_lock()
+        type StakingLockId: Get<LockIdentifier>;
+        /// Max number of unbond request supported by queue
+        type MaxChunkUnlock: Get<usize>;
+        /// The origin which can cancel a deferred slash. Root can always do this.
+        type CancelOrigin: EnsureOrigin<Self::Origin>;
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(crate) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            if Self::do_upgrade_pallet() {
+                DoUpgradePallet::<T>::put(false);
```

Comment:
When we will deploy the runtime upgrade this variable will be default be set to `false`, which means the runtime upgrade will not happen. Maybe a viable option would be to use the recently introduced pallet versioning feature? There is an example here capsule-corp-ternoa/chain#53

Reply:
Done Eliott :+1: